src/snowflake/snowpark/_internal/open_telemetry_prototype.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.
+#
+
+import inspect
+from typing import Dict
+
+# this parameter make sure no error when open telemetry is not installed
+open_telemetry_not_found = False
+try:
+    from opentelemetry import trace
+    from opentelemetry.sdk.resources import SERVICE_NAME, Resource
+
+except ImportError:
+    open_telemetry_not_found = True
+
+if not open_telemetry_not_found:
+    resource = Resource(attributes={SERVICE_NAME: "snowpark-python-open-telemetry"})
+    tracer = trace.get_tracer("snowflake.snowpark.dataframe")
+
+
+def open_telemetry(name):
+    def open_telemetry_decorator(func):
+        def wrapper(*df, **params):
+            # get complete parameter list of the function
+            dataframe, parameters = parameter_decoder(df, params, func, name)
+            with tracer.start_as_current_span(name) as cur_span:
+                # store parameters passed into action function in span
+                # remove for now, will clean up with final design
+                # cur_span.set_attribute("function_parameters", str(parameters))
+
+                # store execution location in span
+                frame_info = inspect.stack()[-1]
+                cur_span.set_attribute("code.filepath", f"{frame_info.filename}")
+                cur_span.set_attribute("code.lineno", f"{frame_info.lineno}")
```

Comment:
Protect computation of the attributes with https://opentelemetry-python.readthedocs.io/en/latest/api/trace.span.html#opentelemetry.trace.span.Span.is_recording